language/resource-api/README.md

```diff
@@ -356,6 +356,43 @@ For example, in the Puppet runtime environment this is bound to the `strict` set
 ```
 
 
+### Provider feature: `custom_insync`
+
+Allows for overriding default property `insync?` checking using custom logic in the provider.
+
+```ruby
+# lib/puppet/type/apt_key.rb
+Puppet::ResourceApi.register_type(
+  name: 'apt_key',
+  features: [ 'custom_insync' ],
+)
+
+# lib/puppet/provider/apt_key/apt_key.rb
+class Puppet::Provider::AptKey::AptKey
+  def insync?(context, name, attribute_name, is_hash, should_hash)
+    case attribute_name
+    when :some_array_attribute
+      if should_hash[:force]
```

Comment:
please do define the attributes used here in the type, so that it is clear how they are defined (property vs parameter).

Since this is also only called for properties, it's `property_name`, not `attribute_name` in the sig. Also needs to be changed in the test in https://github.com/puppetlabs/puppet-resource_api/pull/285